### PR TITLE
Upgrade to version 3 of Terraform AWS provider

### LINKED ISSUE #7: Automate association of a VPC with a private DNS zone (cross-account)
Terraform is currently unable to automate the process of associating a VPC in one AWS account with a private DNS zone in a different AWS account.  This [feature request](https://github.com/terraform-providers/terraform-provider-aws/issues/384) details the issue.

This process can be done via the AWS CLI, so that is [our current manual workaround](https://github.com/cisagov/cool-assessment-terraform/blob/develop/vpc.tf#L40-L63):
```
aws route53 create-vpc-association-authorization \
--hosted-zone-id COOLPRIVATEZONEID \
--vpc VPCRegion=us-east-1,VPCId=vpc-MY_ENV_VPC_ID \
--profile cool-sharedservices-provisionaccount

aws route53 associate-vpc-with-hosted-zone \
--hosted-zone-id COOLPRIVATEZONEID \
--vpc VPCRegion=us-east-1,VPCId=vpc-MY_ENV_VPC_ID \
--profile cool-MYENV-provisionaccount
```

It may be possible to use a Terraform module (such as [this one](https://github.com/opetch/terraform-aws-cli-resource)) to accomplish this automatically until Terraform is able to natively perform this action.

### PR DESCRIPTION
**Pull request by dav3r**
# 

## 🗣 Description
This PR moves this repo from using version 2 to version 3 of the [Terraform AWS provider](https://github.com/terraform-providers/terraform-provider-aws).

## 💭 Motivation and Context
Version 3 of the AWS provider appears to have stabilized, so it seems like a good time for us to jump on board.

It will also allow us to resolve https://github.com/cisagov/cool-assessment-terraform/issues/7 by taking advantage of https://github.com/terraform-providers/terraform-provider-aws/pull/14215.

I will keep this PR as a draft until https://github.com/cisagov/cert-read-role-tf-module/pull/19 is merged.  After that, I will revert https://github.com/cisagov/cool-assessment-terraform/commit/c25b9d4cde1d208002f409cc574c494facf7c0bf, finalize this PR, and request review.

## 🧪 Testing
* All automated tests pass.
* The Terraform in this repo was successfully applied and destroyed in a Staging environment (while [using the version 3 branch of cert-read-role-tf-module](https://github.com/cisagov/cool-assessment-terraform/commit/c25b9d4cde1d208002f409cc574c494facf7c0bf)).

## 🚥 Types of Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (causes existing functionality to change)

## ✅ Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: guacamole_iam.tf [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # Create a role that allows the instance to read its certs from S3.
 module "guacamole_certreadrole" {
-  source = "github.com/cisagov/cert-read-role-tf-module"
+  source = "github.com/cisagov/cert-read-role-tf-module?ref=test_tf_aws_provider_v3"
```
Comment (jsf9k) [inline guacamole_iam.tf]: Don't forget to revert this before merging.

**Review by jsf9k — APPROVED** [review]
Huzzah!

File: guacamole_iam.tf (hunk 1 above)
Comment (dav3r) [reply guacamole_iam.tf]: Reverted with https://github.com/cisagov/cool-assessment-terraform/pull/69/commits/e2e15c3783a5b48d0ac2dc3b2d1d3f1851d3cc3d.

**Review by hillaryj — APPROVED** [review]
🥉

**Review by felddy — APPROVED** [review]
Short and sweet.  🩳 🍬

**Pull request merged**